src/components/modals/FlipInputModal2.tsx

```diff
@@ -0,0 +1,301 @@
+import { ExchangedFlipInput2, ExchangedFlipInputAmounts, ExchangedFlipInputRef, ExchangeFlipInputFields } from '../themed/ExchangedFlipInput2'
+import { MiniButton } from '../themed/MiniButton'
+import { ThemedModal } from '../themed/ThemedModal'
+
+export type FlipInputModalResult = {
+  nativeAmount: string
+  exchangeAmount: string
+}
+
+export interface SetFeesParams {
+  feeTokenId?: string
+  feeNativeAmount: string
+}
+export interface FlipInputModalRef {
+  setFees: (params: SetFeesParams) => void
+  setAmount: (field: ExchangeFlipInputFields, value: string) => void
+  setError: (errorMessage: string | null) => void
+}
+
+type FeeStyleTypes = 'dangerText' | 'warningText'
+
+type Props = {
+  bridge: AirshipBridge<FlipInputModalResult>
+  walletId: string
+  tokenId?: string
+  startNativeAmount?: string
+  forceField?: ExchangeFlipInputFields
+  // Fees
+  feeTokenId?: string
+  feeNativeAmount?: string
+  feeStyle?: FeeStyleTypes
+  onFeesChange?: () => void
+  onMaxSet?: () => void
+  onAmountsChanged?: (amounts: ExchangedFlipInputAmounts) => void
+  headerText?: string
+  hideMaxButton?: boolean
+}
+
+const FlipInputModal2Component = React.forwardRef<FlipInputModalRef, Props>((props: Props, ref) => {
+  const {
+    bridge,
+    walletId,
+    tokenId,
+    startNativeAmount,
+    forceField,
+    onAmountsChanged,
+    feeTokenId: startingFeeTokenId,
+    feeNativeAmount: startingFeeNativeAmount = '',
+    feeStyle,
+    onFeesChange,
+    onMaxSet,
+    headerText,
+    hideMaxButton
+  } = props
+
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatch(account, 'currencyWallets')
+  const [feeTokenId, setFeeTokenId] = useState<string | undefined>(startingFeeTokenId)
+  const [feeNativeAmount, setFeeNativeAmount] = useState<string>(startingFeeNativeAmount)
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+  const wallet = currencyWallets[walletId]
+  const { currencyInfo } = wallet
+  const balances = useWatch(wallet, 'balances')
+  const { pluginId } = currencyInfo
+  const currencyCode = getCurrencyCode(wallet, tokenId)
+  const displayDenom = useDisplayDenom(pluginId, currencyCode)
+  const exchangedFlipInputRef = React.useRef<ExchangedFlipInputRef>(null)
+  const feeCurrencyCode = getCurrencyCode(wallet, feeTokenId)
+  const feeDisplayDenom = useDisplayDenom(pluginId, feeCurrencyCode)
+
+  const [amounts, setAmounts] = useState<ExchangedFlipInputAmounts>({
+    nativeAmount: '',
+    exchangeAmount: '',
+    fiatAmount: '',
+    fieldChanged: forceField ?? 'fiat'
+  })
+
+  const walletName = wallet?.name ?? s.strings.string_no_name
+  const flipInputHeaderText = headerText ?? sprintf(s.strings.send_from_wallet, walletName)
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const handleAmountsChanged = useHandler((amounts: ExchangedFlipInputAmounts) => {
+    setAmounts(amounts)
+
+    if (onAmountsChanged) onAmountsChanged(amounts)
```

Comment:
Please use `onAmountsChanged != null`. While we don't currently have the strict boolean lint rule switched on, we want to move towards getting it enabled one day.

Reply:
done